### PR TITLE
Details about AutomaticDecompression

### LINKED ISSUE #3742: HttpClientHandler.AutomaticDecompression only decompresses one layer
**Issue description**

In HttpClientHandler, you are allowed to set AutomaticDecompression (values are GZip, Deflate, Brotli, None, or a combination of those). The idea is that .NET will automatically decompress HTTP responses for you.

Compression on the HTTP response body is defined by by the "Content-Encoding" header. However, this can have multiple values if the content was compressed multiple times:

```Content-Encoding: gzip, br```

They are listed in the order that they were applied. That is, this server applied GZip compression to some content, and then applied Brotli compression to that gzipped content.

Both the .NET Framework and .NET Core source code have a peculiar behavior though: if there are multiple specified, it only decompresses the last - or outermost - compression. So in this case, it would decompress the Brotli-encoded content, and then pass along the GZipped content to the user. It would also strip the word "brotli" from the Content-Encoding header, so the user would se it as ```Content-Encoding: gzip```.

[Link to .NET Core source code that does this](https://github.com/dotnet/runtime/blob/35fe82c2e90fa345d0b6ad243c14da193fb123fd/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs#L63)

[Link to .NET Framework source code that does this](https://github.com/microsoft/referencesource/blob/17b97365645da62cf8a49444d979f94a59bbb155/System/net/System/Net/HttpWebResponse.cs#L540)

(I also made a small repro, ask if you need it)

As far as I know this isn't *incorrect* - .NET is allowed to limit itself to 1 decompression, I guess - but it's not documented anywhere. The right place might be in the docs for HttpClientHandler, or HttpClientHandler.AutomaticDecompression. Though it would also be nice if the .NET team provided a reason why .NET chooses to do this.

**Target framework**

All versions of Core/Framework/Standard, as far as I know.

### PR DESCRIPTION
**Pull request by ManickaP**
Fixes #3742

### CODE REVIEW

File: xml/System.Net.Http/HttpClientHandler.xml [hunk 1]
```diff
@@ -207,6 +207,12 @@ These configuration options are not available starting with .NET 5.
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+ Automatic decompression works only on the top most compression. So if the content is compressed mutliple times, only the last compression will be decompressed.
```
Comment (liveans) [inline xml/System.Net.Http/HttpClientHandler.xml:210]:
```suggestion
 Automatic decompression works only on the top most compression. So if the content is compressed multiple times, only the last compression will be decompressed.
```

File: xml/System.Net.Http/SocketsHttpHandler.xml [hunk 2]
```diff
@@ -180,7 +180,17 @@ These configuration options are not available starting with .NET 5.
       <Docs>
         <summary>Gets or sets the type of decompression method used by the handler for automatic decompression of the HTTP content response.</summary>
         <value>The type of decompression method used by the handler for automatic decompression of the HTTP content response.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+ Automatic decompression works only on the top most compression. So if the content is compressed mutliple times, only the last compression will be decompressed.
```
Comment (liveans) [inline xml/System.Net.Http/SocketsHttpHandler.xml:187]:
```suggestion
 Automatic decompression works only on the top most compression. So if the content is compressed multiple times, only the last compression will be decompressed.
```

**Review by liveans — APPROVED** [review]
LGTM!

File: xml/System.Net.Http/HttpClientHandler.xml [hunk 3]
```diff
@@ -207,6 +207,12 @@ These configuration options are not available starting with .NET 5.
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+ Automatic decompression works only on the top most compression. So if the content is compressed multiple times, only the last compression will be decompressed.
+ For example, for a reponse with `Content-Encoding: gzip, br` header, the content will only be decompressed using Brotli, leaving the still gzipped content in the <xref:System.Net.Http.HttpResponseMessage.Content>.
```
Comment (stephentoub) [inline xml/System.Net.Http/HttpClientHandler.xml:211]: > reponse

response

Comment (stephentoub) [inline xml/System.Net.Http/HttpClientHandler.xml:211]: > For example, for a reponse with `Content-Encoding: gzip, br` header

"For example, for a response with a `Content-Encoding: gzip, br` header that contains content first compressed with gzip and then that compressed content Brotli compressed,"

File: xml/System.Net.Http/HttpClientHandler.xml [hunk 4]
```diff
@@ -207,6 +207,12 @@ These configuration options are not available starting with .NET 5.
...
+
+ Setting automatic decompression to anything else than <xref:System.Net.DecompressionMethods.None> will cause adding `Accept-Encoding` header with the set values into every outgoing <xref:System.Net.Http.HttpRequestMessage>.
```
Comment (stephentoub) [inline xml/System.Net.Http/HttpClientHandler.xml:213]: > to anything else than

"to anything other than"

Comment (stephentoub) [inline xml/System.Net.Http/HttpClientHandler.xml:213]: > will cause adding `Accept-Encoding` header with

"will result in adding an `Accept-Encoding` header with"

File: xml/System.Net.Http/SocketsHttpHandler.xml [hunk 5]
```diff
@@ -180,7 +180,17 @@ These configuration options are not available starting with .NET 5.
       <Docs>
         <summary>Gets or sets the type of decompression method used by the handler for automatic decompression of the HTTP content response.</summary>
         <value>The type of decompression method used by the handler for automatic decompression of the HTTP content response.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
```
Comment (stephentoub) [inline xml/System.Net.Http/SocketsHttpHandler.xml:186]: Same comments as above.

**Pull request merged**